### DISCUSSION TITLE
Use with NextJS, where can I put the loadAllLocales function?

### DISCUSSION THREAD

**Discussion by IRediTOTO** [ivanhofer/typesafe-i18n#402] [Q&A] [answered]
Hi, I use `typesafe-i18n` with Nextjs, everything work fine. But new version I don't know the right way to do that. Can you please tell me where to put the function `loadAllLocales()` ?
It only work if I put in file `_app.js` and outside of component App

```
import { loadAllLocales } from 'src/i18n/i18n-util.sync';

loadAllLocales(); // Where can I put this ?

const App = ({ Component, pageProps, router }: AppProps) => {
.....
}

```

**Comment by ivanhofer** [comment, marked as answer]
Please take a look at the official Next.js example: https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/adapter-react/examples/nextjs
You  should not call `loadAllLocales`, instead you should use `loadLocaleAsync` and  just load the locale your user needs.

**Reply by IRediTOTO** [reply]
So I must call `GetStaticProps` on every page? This look like so tired :(

**Reply by ivanhofer** [reply]
yes, it looks so. I also don't like it but it seems Next.js can only be used in that way. I tried to make an official example already a few times and I always gave up because I didn't like the result to make SSR work. I'm not an Next.js expert, so maybe there is better solution but nobody I talked to knows a better way.
